src/transformers/trainer.py

```diff
@@ -408,7 +417,7 @@ def get_eval_dataloader(self, eval_dataset: Optional[Dataset] = None) -> DataLoa
         """
         Returns the evaluation :class:`~torch.utils.data.DataLoader`.
 
-        Will use no sampler if :obj:`self.eval_dataset` is a :obj:`torch.utils.data.IterableDataset`, a sequential
+        Will use no sampler if :obj:`self.eval_dataset` does not implement :obj:`__len__`, a sequential
```

Comment:
This is not the behavior (it raises an error) so this should be documented this way.